DxDispatch/src/dxdispatch/Executor.cpp

```diff
@@ -158,18 +172,32 @@ void Executor::operator()(const Model::DispatchCommand& command)
             }
 
             PIXBeginEvent(m_device->GetCommandQueue(), PIX_COLOR(255, 255, 0), "Dispatch '%s'", command.dispatchableName.c_str());
-            
+
+            uint32_t timestampIndex = (iteration % timestampCount) * 2;
+
+            m_device->GetCommandList()->EndQuery(m_device->GetQueryHeap(), D3D12_QUERY_TYPE_TIMESTAMP, timestampIndex);
             dispatchable->Dispatch(command);
+            m_device->GetCommandList()->EndQuery(m_device->GetQueryHeap(), D3D12_QUERY_TYPE_TIMESTAMP, timestampIndex + 1);
+            m_device->GetCommandList()->ResolveQueryData(m_device->GetQueryHeap(), D3D12_QUERY_TYPE_TIMESTAMP, timestampIndex, 2, m_timestampResultBuffers, timestampIndex * sizeof(UINT64));
```

Comment:
Rather than resolving query data within the loop, why not resolve it after the loop has finished? You just need the query heap and results buffer to be the same size, then you can resolve all of the timestamps in one shot. It's best to minimize additional profiling work done in the loop since it can marginally affect timing results.